protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/dependencies/aether/AetherMavenArtifactPathResolver.java

```diff
@@ -159,17 +166,34 @@ public Collection<Path> resolveDependencies(
       var scopeFilter = new ScopeDependencyFilter(dependencyScopes);
       var collectRequest = new CollectRequest(dependenciesToResolve, null, remoteRepositories);
       var dependencyRequest = new DependencyRequest(collectRequest, scopeFilter);
-
-      return repositorySystem.resolveDependencies(repositorySession, dependencyRequest)
-          .getArtifactResults()
-          .stream()
-          .map(ArtifactResult::getArtifact)
-          .map(this::determinePath)
-          .collect(Collectors.toUnmodifiableList());
+      dependencyResult = repositorySystem.resolveDependencies(repositorySession, dependencyRequest);
 
     } catch (DependencyResolutionException ex) {
-      throw new ResolutionException("Failed to resolve dependencies", ex);
+      // GH-299: if this exception is raised, we may still have some results we can use. If this is
+      // the case then resolution only partially failed, so continue for now unless strict
+      // resolution is enabled. We do not fail-fast here anymore (since 2.4.0) as any resolution
+      // errors should be dealt with by the maven-compiler-plugin later on if needed.
+      //
+      // If we didn't get any result, then something more fatal has occurred, so raise.
+      dependencyResult = ex.getResult();
+
+      if (dependencyResult == null || failOnInvalidDependencies) {
+        throw new ResolutionException("Failed to resolve dependencies", ex);
+      }
+
+      // Log the message as well here as we omit it by default if `--errors' is not passed to Maven.
+      log.error(
```

Comment:
```suggestion
      log.warn(
```

Reduce to a warning to match the Maven behaviour